File: src/ELibrary.Frontend/src/app/modules/shared/domain/dtos/shop/statistics/getShopStatisticsRequest.ts
import { GetShopStatistics, mapBookToStatisticsBook, StatisticsBook } from "../../../..";

export interface GetShopStatisticsRequest {
    fromUTC: Date | null,
    toUTC: Date | null,
    includeBooks: StatisticsBook[]
}

export function mapGetShopStatisticsToGetShopStatisticsRequest(statistics: GetShopStatistics): GetShopStatisticsRequest {
    return {
        fromUTC: statistics.fromUTC,
        toUTC: statistics.toUTC,
        includeBooks: statistics.includeBooks.map(x => mapBookToStatisticsBook(x))
    }
}

File: src/ELibrary.Backend/ShopApi/Features/StatisticsFeature/Domain/Dtos/GetShopStatisticsRequest.cs
namespace ShopApi.Features.StatisticsFeature.Domain.Dtos
{
    public class GetShopStatisticsRequest
    {
        public DateTime? FromUTC { get; set; }
        public DateTime? ToUTC { get; set; }
        public StatisticsBookRequest[] IncludeBooks { get; set; }
    }
}
